src/components/PopupWithForm.js

```diff
@@ -2,32 +2,44 @@ import Popup from "./Popup.js";
 
 export default class PopupWithForm extends Popup {
   constructor(popupSelector, handleFormSubmit) {
-    super({popupSelector});
+    super({ popupSelector });
     this._handleFormSubmit = handleFormSubmit;
-    this._form = this._popupElement.querySelector('.modal__form');
+    this._form = this._popupElement.querySelector(".modal__form");
   }
 
   _getInputValues() {
     const inputValues = {};
     this._inputList = this._form.querySelectorAll(".modal__input");
-    this._inputList.forEach(input => {
-      inputValues[input.name] = input.value;  //(???)
+    this._inputList.forEach((input) => {
+      inputValues[input.name] = input.value; //(???)
     });
     return inputValues;
   }
 
-  setEventListeners(){
+  setEventListeners() {
     super.setEventListeners();
-    this._form.addEventListener('submit', (evt) => {
+    this._form.addEventListener("submit", (evt) => {
       evt.preventDefault();
       this._handleFormSubmit(this._getInputValues());
       this.close();
     });
   }
 
-close() {
+  close() {
     super.close();
   }
 
-}
+  setSubmit(handleSubmit) {
+    this._handleFormSubmit = handleSubmit;
+  }
 
+  renderLoading(isSaving) {
+    this._modalButton = this._popupElement.querySelector(".modal__button");
```

Comment:
**Could be improved**

All elements that are found every time you call a method should be defined as **class fields** in the constructor, `_setEventListeners` (or in `getView`) only **1 time** not to waste resources on searching them again and again.